source/material_model/melt_simple.cc

```diff
@@ -258,97 +236,110 @@ namespace aspect
-                            reaction_rate_out->reaction_rates[i][c] = out.reaction_terms[i][c] / melting_time_scale;
-                          else if (c == porosity_idx && this->get_timestep_number() > 0)
-                            reaction_rate_out->reaction_rates[i][c] = melting_rate / melting_time_scale;
-                          else
-                            reaction_rate_out->reaction_rates[i][c] = 0.0;
-                        }
-                      out.reaction_terms[i][c] = 0.0;
+                      if (c == peridotite_idx && this->get_timestep_number() > 0)
+                        reaction_rate_out->reaction_rates[i][c] = porosity_change / melting_time_scale
+                                                                  * (1 - maximum_melt_fraction) / (1 - old_porosity);
+                      else if (c == porosity_idx && this->get_timestep_number() > 0)
+                        reaction_rate_out->reaction_rates[i][c] = porosity_change / melting_time_scale;
+                      else
+                        reaction_rate_out->reaction_rates[i][c] = 0.0;
                     }
+                  out.reaction_terms[i][c] = 0.0;
                 }
 
               const double porosity = std::min(1.0, std::max(in.composition[i][porosity_idx],0.0));
               out.viscosities[i] = eta_0 * exp(- alpha_phi * porosity);
+
+              out.entropy_derivative_pressure[i]    = entropy_change (in.temperature[i], this->get_adiabatic_conditions().pressure(in.position[i]), maximum_melt_fraction, NonlinearDependence::pressure);
+              out.entropy_derivative_temperature[i] = entropy_change (in.temperature[i], this->get_adiabatic_conditions().pressure(in.position[i]), maximum_melt_fraction, NonlinearDependence::temperature);
             }
           else
             {
               out.viscosities[i] = eta_0;
 
+              out.entropy_derivative_pressure[i]    = entropy_change (in.temperature[i], this->get_adiabatic_conditions().pressure(in.position[i]), 0, NonlinearDependence::pressure);
+              out.entropy_derivative_temperature[i] = entropy_change (in.temperature[i], this->get_adiabatic_conditions().pressure(in.position[i]), 0, NonlinearDependence::temperature);
+
               // no melting/freezing is used in the model --> set all reactions to zero
               for (unsigned int c=0; c<in.composition[i].size(); ++c)
                 {
                   out.reaction_terms[i][c] = 0.0;
 
-                  if (this->get_parameters().use_operator_splitting && reaction_rate_out != NULL)
+                  if (reaction_rate_out != NULL)
                     reaction_rate_out->reaction_rates[i][c] = 0.0;
                 }
             }
 
-          out.entropy_derivative_pressure[i]    = entropy_change (in.temperature[i], this->get_adiabatic_conditions().pressure(in.position[i]), maximum_melt_fractions[i], NonlinearDependence::pressure);
```

Comment:
What happened to these lines? Are they not important for latent heat?

Reply:
I moved them further up into the individual branches (with and without melt transport). This was convenient because they need the depletion field, which we do not necessarily have in the case without melt, and so we compute it only in the part of the if statement that is executed with melt transport (otherwise, 0 is used, just as before these changes).